### PR TITLE
Move OAuth Callback logic to separate hook

### PR DESCRIPTION
**Pull request by DiD3n**

### CODE REVIEW

File: src/api/rest/auth/oauth/OAuthProviders.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
-enum Provider {
+enum OAuthProviders {
```
Comment (JUtupe) [inline src/api/rest/auth/oauth/OAuthProviders.ts]: use singular form here

File: src/contexts/auth/hooks/useGenericOAuthCallback.hook.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    success,
+    error,
+}
+
+export interface OAuthCallbackRet {
+    error: string|null
+    status: OAuthStatus
+}
+
+export const useGenericOAuthCallback = (provider: OAuthProviders): OAuthCallbackRet => {
```
Comment (JUtupe) [inline src/contexts/auth/hooks/useGenericOAuthCallback.hook.ts]: This name would be better without the `generic` part

**Review by JUtupe — CHANGES_REQUESTED** [review]

File: src/api/rest/auth/oauth/OAuthProviders.ts (hunk 1 above)
Comment (DiD3n) [reply src/api/rest/auth/oauth/OAuthProviders.ts]: 👍

**Review by JUtupe — APPROVED** [review]

**Review by JUtupe — APPROVED** [review]
Next time when you add new changes after review, try to do it in another PR

**Pull request merged**